compiler/rustc_builtin_macros/src/assert/context.rs

```diff
@@ -36,9 +66,249 @@ impl<'cx, 'a> Context<'cx, 'a> {
     ///        );
     ///    }
     /// }
-    pub(super) fn build(self, _cond_expr: P<Expr>, _panic_path: Path) -> P<Expr> {
-        let Self { cx, span, .. } = self;
-        let stmts = Vec::new();
+    pub(super) fn build(mut self, mut cond_expr: P<Expr>, panic_path: Path) -> P<Expr> {
+        let expr_str = pprust::expr_to_string(&cond_expr);
+        let mut fmt_str = String::new();
```

Comment:
why are some things mutable fields of `Self`, but this is not?

Reply:
Good question 🤷

Just moved `fmt_str` to Self.